cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggeringInfoHelper.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.schedule.trigger;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.cdap.cdap.api.schedule.Trigger;
+import io.cdap.cdap.api.schedule.TriggerInfo;
+import io.cdap.cdap.internal.app.runtime.schedule.DefaultTriggeringScheduleInfo;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.TriggeringInfo;
+import io.cdap.cdap.proto.TriggeringPropertyMapping;
+import io.cdap.cdap.proto.id.ProgramRunId;
+import io.cdap.cdap.proto.id.ScheduleId;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helper functions for {@link TriggeringInfo}
+ */
+public class TriggeringInfoHelper {
+
+  private static Gson gson = new Gson();
+  private static final String TRIGGERING_PROPERTIES_MAPPING = "triggering.properties.mapping";
+
+  /**
+   * Create a new {@link TriggeringInfo} object given
+   * {@link DefaultTriggeringScheduleInfo}, {@link ScheduleId} and {@link Trigger.Type}
+   * @param triggeringScheduleInfo
+   * @param type
+   * @param scheduleId
+   * @return
+   */
+  public static TriggeringInfo fromTriggeringScheduleInfo(DefaultTriggeringScheduleInfo triggeringScheduleInfo,
```

Comment:
Why take `DefaultTriggeringScheduleInfo` instead of the interface `TriggeringScheduleInfo`? Generally it is preferable to pass around interface.